sdks/java/core/src/main/java/org/apache/beam/sdk/util/WeightedValue.java

```diff
@@ -47,4 +49,26 @@ public long getWeight() {
   public T getValue() {
     return value;
   }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof WeightedValue)) {
```

Comment:
check for null?

Reply:
`null instanceof X` is always false so checking for null is redundant.